### ISSUE TITLE
IPLDMRDTDAG

### ISSUE THREAD

**Issue by goozang-records** [4meta5/consensus#14]
structure a state machine that formats transactions in a way that provides domain-specific context with IPLD.

### calculating dependence

The transaction format should allow us to calculate some meaningful measure of independence between transactions. This dependence measure can be used to manage the flow rate of proposals, voter participation, and other signalling expectations that the blockchain will need to be live to receive and incorporate into its future state. 

### dependence allows us to construct a flexible DAG

The metric for dependence does not need to be binary -- the point is to use measurements of dependence to allow for stateful merging of transactions while also limiting transaction throughput for certain operations to manage flow rate of actions.

## in summary

When it comes to thinking about the transaction format, I have a few unrelated concepts that I think might prove useful for fostering certain stateful network operations over the read uncommitted (dirty) transactions
* https://ipld.io
* http://kcsrk.info/papers/oopsla19-mrdt.pdf

I'd also like to aspire to one of age's goal and adopt a simple, streaming encryption protocol (like noise/disco): "*A good seekable streaming encryption scheme based on modern chunked AEADs, reusable as a general encryption format*", but this should probably be done in some client-side library (the node)
